packages/libs/tool-sdk/src/lib/assertSupportedToolVersion.ts

```diff
@@ -0,0 +1,18 @@
+import { major } from 'semver';
+
+import { VINCENT_TOOL_API_VERSION } from './constants';
+
+export function assertSupportedToolVersion(
+  toolVersionSemver: string | undefined,
+): asserts toolVersionSemver is string {
+  if (!toolVersionSemver) {
+    throw new Error('Tool version is required');
+  }
+
+  // Check if the tool's API version has the same major version as the current API version
+  if (major(toolVersionSemver) !== major(VINCENT_TOOL_API_VERSION)) {
```

Comment:
I think the check should also include >= for minor and patch when necessary

The tool might require a new feature added at a certain minor or a bug fix, which are not major changes